File: C++/4.1.CountingElements_PermCheck.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int solution(vector<int> &A){
    sort(vector.begin(), vector.end());
    for( int i = 0; i < A.size(); i++){
        if( i+1 != A[i] ){
            return 0;
        } 
    }
    return 1;
}


int main(){
return 0;
}

// O(N) or O(N * log(N)) -> 100%

File: Python/4.1.CountingElements_PermCheck.py
def solution(A):
    A.sort()
    for i in range(len(A)):
        if i+1!=A[i]:
            print("i is", i)
            return(0)
    return(1)

A = [i for i  in range(1,100000)]
print(solution(A))


# O(N) or O(N*log(N)) -> 100%

